From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Tue, 28 Jul 2026 00:41:18 +0200
Subject: arm64: dts: allwinner: sun50i-a64-pinephone: Add speaker amplifier
 supply

The speaker amplifier is powered from the PS rail (the PMIC power
source output, modeled by the always-on ps-regulator). Without the
supply the driver uses a dummy regulator:

  simple-amplifier audio-amplifier: supply VCC not found, using dummy
  regulator

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
@@ -455,6 +455,7 @@ speaker_amp: audio-amplifier {
 		compatible = "simple-audio-amplifier";
 		enable-gpios = <&pio 2 7 GPIO_ACTIVE_HIGH>; /* PC7 */
 		sound-name-prefix = "Speaker Amp";
+		VCC-supply = <&reg_ps>;
 	};
 
 	vibrator {
-- 
Armbian

